arch/arm/src/armv7-a/arm_cache.c

```diff
@@ -190,7 +198,15 @@ void up_clean_dcache_all(void)
 
 void up_flush_dcache(uintptr_t start, uintptr_t end)
 {
-  cp15_flush_dcache(start, end);
+  if (cp15_cache_size() < (end - start))
```

Comment:
I have a question about how optimal is current implementation if memory from `start` till `end` is not currently in cache? Or some part is in cache and other part is not?

Reply:
We have a testing on armv8-m with 32K icache & 32k dcache, when flush &clean buffer with size 1MB。

up_clean_dcache_all() run 30 times faster than up_clean_dcache().

When use this patch, the time is same.